src/main/java/seedu/address/model/task/Recurring.java

```diff
@@ -0,0 +1,140 @@
+       this.type = parseRecurType(recurringString);
+       this.referenceDateTime = referenceDateTime;
+   }
+
+   public LocalDateTime parseDateTime(String dateTimeString) {
+       return stringFormatter.parse(dateTimeString, LocalDateTime::from);
+   }
+
+   public RecurType parseRecurType(String recurringString) throws ParseException {
+       if (recurringString.equals("d")) {
+           return RecurType.DAILY;
+       } else if (recurringString.equals("w")) {
+           return RecurType.WEEKLY;
+       } else {
+           throw new ParseException(Recurring.MESSAGE_CONSTRAINTS);
+       }
+   }
+   
+   /**
+    * Returns a copy of the task that is being reset with everything being the same except for whether the method is Done
+    * @param taskToReset
+    * @return copied task with done set to undone
+    */
+   public Task resetDone(Task taskToReset) {
+       assert taskToReset != null;
+
+        Name updatedName = taskToReset.getName();
+        Priority updatedPriority = taskToReset.getPriority();
+        Description updatedDescription = taskToReset.getDescription();
+        Set<Tag> updatedTags = taskToReset.getTags();
+        Optional<Reminder> sameOptReminder = taskToReset.getOptionalReminder();
+        Optional<Recurring> sameOptRecurring = taskToReset.getOptionalRecurring();
+        return new Task(
+                updatedName, updatedPriority, updatedDescription, new Done("N"), updatedTags, sameOptReminder, sameOptRecurring);
+   }
+
+   /** Returns true if a given string is a valid name. */
+   public static boolean isValidRecurring(String test) {
+       return test.matches(VALIDATION_REGEX);
+   }
+   
+   /**
+    * Returns Daily or Weekly for display on the card.
+    */
+   public String displayRecurring() {
+       String result = type.name().substring(0, 1) + type.name().substring(1).toLowerCase();
+       return result;
+   } 
+
+
+   public TimerTask generateTimerTask(Model model, Task taskToReset) {    
+    
+    return new TimerTask(){
+        @Override
+        public void run() {
+            Platform.runLater(() -> {
+                requireNonNull(model);
+
+                Task resetedTask = resetDone(taskToReset);
+
+                model.setTask(taskToReset, resetedTask);
+                model.updateFilteredTaskList(PREDICATE_SHOW_ALL_TASKS);
+                }
+            );
+            
+        }
+    };
+   }
+
+   public void triggerRecurring(Model model, Task taskToReset) {
+       TimerTask repeatedTask = generateTimerTask(model, taskToReset);
+       Timer timer = new Timer("Timer");
+       long period = getInterval();
+       long delayToFirstTrigger = Duration.between(LocalDateTime.now(), referenceDateTime).getSeconds();
+       delayToFirstTrigger = delayToFirstTrigger >= 0 ? delayToFirstTrigger*1000 : 0;
+       timer.scheduleAtFixedRate(repeatedTask, delayToFirstTrigger, period); //might run twice in the first time
```

Comment:
if timer is not part of the class attributes, will it reschedule after the end of the function call?

Reply:
While the application is open, the timer will call reschedule at the fixed rate in the time delay. 
If the application is closed, it will triggerRecurring for all the tasks that have it in the getFilteredTaskList method call